add SidedScrollOptionBehavior and made AllIcons usable with custom resource

slin-o opened the pull request:
added new behavior that acts like ScrollOptionBehavior that stores values per side instead of for the enttire block

added new constructor to AllIcons to create custom icons with your own icon texture

slin-o committed:
add SidedScrollOptionBehavior and made AllIcons usable with custom rescource

slin-o committed:
fix permissions on gradlew file

slin-o committed:
fix Client not recognizing sided behavior on opening menu

VoidLeech commented:
Is the AllIcons change related to the scrolloptionbehaviour? That might be better in a separate PR if not.

slin-o commented:
> Is the AllIcons change related to the scrolloptionbehaviour? That might be better in a separate PR if not.

the ScrollOptionBehavior needs an INamedIcons enum wich requires an  AllIcons to be returned in one method to render the correct selection

slin-o commented:
why is this the wrong target @VoidLeech

VoidLeech commented:
The earliest supported version atm is 1.20.1

slin-o commented:
ok

TropheusJ commented on src/main/java/com/simibubi/create/foundation/gui/AllIcons.java:
```diff
@@ -21,8 +21,8 @@
 
-	public static final ResourceLocation ICON_ATLAS = Create.asResource("textures/gui/icons.png");
-	public static final int ICON_ATLAS_SIZE = 256;
+	public final ResourceLocation ICON_ATLAS ;
```
these fields should stay static. Add new instance fields that get set to these in the constructor.

slin-o replied:
added instance fields instead in latest commit

TropheusJ commented on src/main/java/com/simibubi/create/foundation/blockEntity/behaviour/scrollValue/ISidedScrollValueBehavior.java:
```diff
@@ -0,0 +1,14 @@
+
+import net.minecraft.core.Direction;
+
+public interface ISidedScrollValueBehavior<B extends ScrollValueBehaviour>{
```
new interfaces should not be I-prefixed

slin-o replied:
you're right, I thought I had a class named SidedScrollValueBehavior, will fix that

slin-o replied:
renamed in latest commit

TropheusJ commented on src/main/java/com/simibubi/create/foundation/blockEntity/behaviour/scrollValue/SidedScrollOptionBehavior.java:
```diff
@@ -0,0 +1,65 @@
+	 * @param be The SmartBlockEntity this behavior is associated with.
+	 * @param slot The ValueBoxTransform for the slot.
+	 */
+
```
weird extra line

slin-o replied:
yeah, maybe a formatting issue or just accidentally left in there

slin-o replied:
removed in latest commit

slin-o committed:
Merge branch 'Creators-of-Create:mc1.21.1/dev' into mc1.21.1/add-sided-scroll-option-behavior

slin-o committed:
remove I-Prefix form SidedScrollValueBehavior Interface

return ICON_ATLAS and ICON_ATLAS_SIZE to static fields and add corresponding instance fields
